src/components/ToDoHeader.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FormEvent, useContext, useRef } from 'react';
+import { DispatchContext, StateContext } from './StateContext';
+import classNames from 'classnames';
+
+export const ToDoHeader = () => {
+  const { toDoTitle, todos, focusOnTodo } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedToDoTitle = toDoTitle.trim();
+
+    if (!trimmedToDoTitle) {
+      return;
+    } else {
+      dispatch({
+        type: 'ADD_TODO',
+      });
+      dispatch({
+        type: 'WRITE_NEW_TITLE',
+        newTitle: '',
+      });
```

Comment:
Reducer is created for this cases u can do those two things with one dispatch so on action `ADD_TODO` you can also clear `toDoTitle`

Reply:
I tried to do this only with reducer and I failed. Should I use useState instead?

Reply:
Did you try to get rid of WRITE_NEW_TITLE action entirely? because you need WRITE_NEW_TITLE in other operations and Sylwester only meant that you can combine dispatches so that ADD_TODO can also clear toDoTitle